llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -2367,9 +2368,29 @@ void computeKnownBits(const Value *V, const APInt &DemandedElts,
   }
 
   // Aligned pointers have trailing zeros - refine Known.Zero set
+  auto SameFunction = [](auto A, auto B) {
+    return A->getParent()->getParent() == B->getParent()->getParent();
+  };
   if (isa<PointerType>(V->getType())) {
     Align Alignment = V->getPointerAlignment(Q.DL);
     Known.Zero.setLowBits(Log2(Alignment));
+    for (auto *User : V->users()) {
```

Comment:
```suggestion
    if (Q.CxtI) {
    Function *CxtFunc = Q.CxtI->getFunction();
    for (auto *User : V->users()) {
```
When CxtI is unavailable, we don't need to iterate the user list.